server/config/config.go

```diff
@@ -991,6 +993,25 @@ func (c *Config) GetZapLogProperties() *log.ZapProperties {
 	return c.logProps
 }
 
+// GetConfigFile gets the config file.
+func (c *Config) GetConfigFile() string {
+	return c.configFile
+}
+
+// RewriteFile rewrites the config file after updating the config.
+func (c *Config) RewriteFile(new *Config) error {
+	filePath := c.GetConfigFile()
+	if filePath == "" {
+		return nil
+	}
+	var buf bytes.Buffer
+	if err := toml.NewEncoder(&buf).Encode(*new); err != nil {
```

Comment:
Looks like it also writes out hidden configurations.

Reply:
Yes.